src/librustc_mir/borrow_check/nll/explain_borrow/mod.rs

```diff
@@ -286,66 +286,62 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
     }
 
     /// Check if a borrow location is within a loop.
-    fn is_borrow_location_in_loop(
+    fn can_reach_ignoring_backedges(
         &self,
         borrow_location: Location,
+        use_location: Location,
     ) -> bool {
         let mut visited_locations = Vec::new();
-        let mut pending_locations = vec![ borrow_location ];
-        debug!("is_in_loop: borrow_location={:?}", borrow_location);
-
-        while let Some(location) = pending_locations.pop() {
-            debug!("is_in_loop: location={:?} pending_locations={:?} visited_locations={:?}",
-                   location, pending_locations, visited_locations);
-            if location == borrow_location && visited_locations.contains(&borrow_location) {
-                // We've managed to return to where we started (and this isn't the start of the
-                // search).
-                debug!("is_in_loop: found!");
+        let mut pending_locations = vec![(borrow_location, false)];
+        debug!(
+            "can_reach_ignoring_backedges: borrow_location={:?}",
+            borrow_location
+        );
+
+        while let Some((location, is_back_edge)) = pending_locations.pop() {
+            if is_back_edge {
+                continue;
+            }
+
+            debug!(
+                "can_reach_ignoring_backedges: location={:?} is_back_edge={:?}
+                   pending_locations={:?} visited_locations={:?}",
+                location, pending_locations, is_back_edge, visited_locations
+            );
+            if location == use_location && visited_locations.contains(&use_location) {
```

Comment:
we don't need the `&&` I don't think, just checking `if location == use_location {` should suffice